blog-post-review.md

```diff
@@ -0,0 +1,261 @@
+
+![](https://hackmd.io/_uploads/r1mQRhz85.png)
+
+
+Now, that's it considering PageRank, next you will see how to use Louvain community detection to find communities in the graph. 
+
+## Community detection with Louvain 
+The Louvain algorithm measures how connected are the nodes within a community if we would compare them to how connected they would be in a random network. Also, it recursively merges communities into a single node and executes the modularity clustering on the condensed graphs. This is one of the most popular community detection algorithms. Let's run it to find how many communities we have an inside graph. First, we will let Louvain execute and save `cluster_id` as a property for every node. 
+
+
+```python
+memgraph.execute(
+    """
+    CALL cugraph.louvain.get() YIELD cluster_id, node
+    SET node.cluster_id = cluster_id;
+    """
+)
+```
+
+Now, let us find number of communities:
+
+```python
+results =  memgraph.execute_and_fetch(
+        """
+        MATCH (n)
+        WITH DISTINCT n.cluster_id as cluster_id
+        RETURN count(cluster_id ) as num_of_clusters;
+        """
+    )
+# we will get only 1 result
+result = list(results)[0]
+
+#don't forget that results are saved in a dict
+print(f"Number of clusters: {result['num_of_clusters']}")
+
+```
+
+    Number of clusters: 2664
+
+
+Next, you can also visualize some of these communities. You can for example find nodes that belong to one community, but are connected to another node that belongs in the opposing community.
+As for Louvain, it tries to minimize that number of nodes, so we shouldn't see that many of them. In Memgraph Lab try to execute the following query:
+
+```
+MATCH  (n2)<-[e1]-
+WHERE n1.cluster_id != m1.cluster_id AND n1.cluster_id = n2.cluster_id
+RETURN *
+LIMIT 1000;
+```
+Here in our query we will `MATCH`  node `n1` and its two relationships `(n2)<-[e1]-(n1)` and `(n1)-[e]->(m1)` but we will filter out only ones `WHERE` cluster_id of nodes `n1` and `n2` is not the same as the  cluster_id of node `m1`. We will return only 1000 such relationships, for easier visualizations.
+
+We used different functions for the `red`, `green`, and `blue` (RGB) color model to create such visualization. From the query above we got the following visualization:
+
+![](https://hackmd.io/_uploads/Syc4weEIq.png)
+
+
+
+
+## Where to next?
+And there you have it - millions of nodes imported and two graph analytics algorithms used. Now you can import huge graphs and do the analytics you want in terms of seconds. Most of all I hope we made those two parts a little easier for you :heart:. If you feel so, don't forget to give us a star on **[Memgraph](https://github.com/memgraph/memgraph)**, **[Memgraph MAGE](https://github.com/memgraph/mage)** and also don't forget to give a star to devs in **[Rapids.ai cuGraph](https://github.com/rapidsai/cugraph)** . :star:
```

Comment:
... and to the analytics,

 you want in a matter of seconds.

Most of all I hope we made those two parts a little easier for you - What two parts exactly?

If you like what we do, don't hesitate to give us a star on bla, bla...